rdflib/graph.py

```diff
@@ -1608,13 +1613,12 @@ def _spoc(
         either triples or quads
         """
         if triple_or_quad is None:
-            return (None, None, None, self.default_context if default else None)
+            return (None, None, None, self.identifier if default else None)
```

Comment:
> This is wrong, isn't it? You want `_spoc` to return a `Graph` for the 4th member.

This specific change is inherited verbatim from the rebased PR [see LoC#1313](https://github.com/RDFLib/rdflib/pull/409/commits/da6837e84235b47a9218a96559a77a33fffe7b89#). Changing it back temporarily doesn't cause any test failures which suggests that if if _is_ wrong, we need a test for it. I added a warning and found the only code that triggers the warning is [`test_conjunctive_graph.py::test_quad_contexts`](https://github.com/RDFLib/rdflib/blob/b2fdaf5a1f45c09694dbd8925ab6b6dee84436b4/test/test_conjunctive_graph.py#L39)

Reply:
The fourth member of the return value of `_spoc` is a `Graph` per the annotation, and it's treated as a `Graph` within this module except in a few cases where I've noted we check whether it is but could more intentionally compare against `None` (and maybe throw a `TypeError` for the case that we don't receive a `Graph` since that is expected)

Reply:
Thanks for the explanation. The collective model of quads seems to have changed somewhat from the original, as documented in the [“Working with multigraphs”](https://github.com/RDFLib/rdflib/blob/master/docs/intro_to_parsing.rst#working-with-multi-graphs) section of the "Introduction to parsing", where the type of the fourth member is explicitly described as an identifier, not a Graph object,

“To read and query multi-graphs, that is RDF data that is context-aware, you need to use rdflib's :class:`rdflib.ConjunctiveGraph` or :class:`rdflib.Dataset` class. These are extensions to :class:`rdflib.Graph` that know all about quads ***(triples + graph IDs)***.” (my emphasis)

The example is equally explicit:
> You could parse the file and query it like this:
```py
from rdflib import Dataset
from rdflib.namespace import RDF

g = Dataset()
g.parse("demo.trig")

for s, p, o, g in g.quads((None, RDF.type, None, None)):
    print(s, g)
```
> This will print out:
```
http://example.com/person/drewp http://example.com/person/graph-1
http://example.com/person/nick http://example.com/person/graph-2
```